fastlane/lib/fastlane/actions/gym.rb

```diff
@@ -24,6 +24,18 @@ def self.run(values)
           # the heavy lifting for them
           values[:export_options] ||= {}
           values[:export_options][:provisioningProfiles] = Actions.lane_context[SharedValues::MATCH_PROVISIONING_PROFILE_MAPPING]
+        elsif Actions.lane_context[SharedValues::SIGH_PROFILE_PATHS]
+          # Since Xcode 9 you need to explicitly provide the provisioning profile per app target
+          # If the user used sigh we can match the profiles from sigh
+          values[:export_options] ||= {}
+          values[:export_options][:provisioningProfiles] ||= {}
+          Actions.lane_context[SharedValues::SIGH_PROFILE_PATHS].each do |profile_path|
```

Comment:
This looks nice, however this code might crash, therefore potentially introducing a regression to existing setups. Can you wrap the inside of the `.each` loop inside a `begin` `rescue` block similarly to what we do at https://github.com/fastlane/fastlane/pull/9460/files#diff-7b900c24bb88ea2776b2b6c4cca7e12cR98? Thanks